### ISSUE TITLE
Improve String handling in SIP crate

### ISSUE THREAD

**Issue by t4lz** [metalbear-co/mirrord#2198]
As pointed out in https://github.com/metalbear-co/mirrord/pull/2195#pullrequestreview-1841106655, we incorrectly use Strings in the SIP crate in places where an `OsString` or a `Vec<u8>` would be appropriate.
I think if we do everything correctly, no error handling should be needed for string conversions, and also nothing like `to_string_lossy`.

There isn't any known issue caused by this yet, but it might be possible to construct an example where this incorrect handling would lead to SIP-sidestepping not working, maybe with some special characters or encoding in file paths or script shebangs.

**Comment by gememma** [comment]
Is there any reason we would need to use `OsString` instead of keeping everything as `PathBuf`? They do things like handle trailing slashes which is currently done manually

**Comment by aviramha** [comment]
Yes, the SIP crate patches/modifies OS-level strings, so to match layout/compatibility we have to use it too.

**Comment by gememma** [comment]
Is `PathBuf` not just a wrapper around `OsString`? Keeping everything as `Path` is still using the underlying `OsString` but with the guarantee that it's always producing a valid path

**Comment by aviramha** [comment]
Ah I didn't now that!
